src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -80,6 +80,24 @@ private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Obje
         return true;
     }
 
+    private static boolean isInstancePrimitive(@Nonnull Class<?> type, @Nonnull Object instance) {
+        if (type.isInstance(instance)) {
+            return true;
+        }
+        // https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.1.2
+        if (instance instanceof Number) {
+            Number n = (Number) instance;
+            if (type == Long.class || type == Double.class) {
+                return true;
+            }
+            if (type == Integer.class && n.longValue() <= Integer.MAX_VALUE && n.longValue() >= Integer.MIN_VALUE) {
+                return true;
+            }
+            // TODO etc. for other types if they ever come up
+        }
+        return false;
+    }
```

Comment:
I think this method could do with some javadoc/comments. For me, the name of the method (`isInstancePrimitive`) does not seem very accurate wrt what it seems like the method is actually doing (checking if `instance` can be widened/narrowed to `type`, right?). `isSafeToConvert(Object from, Class to)`?